mitxpro/views.py

```diff
@@ -43,10 +43,32 @@ def index(request, **kwargs):  # pylint: disable=unused-argument
     """
     The index view
     """
-    if request.method == "GET":
-        return render(request, "index.html", context=get_js_settings_context(request))
-    else:
-        return redirect(request.get_full_path())
+    context = get_js_settings_context(request)
+
+    # pylint: disable=too-many-boolean-expressions
+    if request.method == "POST" and (
```

Comment:
Updated the handler on the frontend to detect if the response to checkout button indicates a `POST` (in case we need to go to CyberSource) or `GET` (in case of $0.0 orders) follow up. In case of $0.0 orders (the order is already fulfilled when the response is received) we send the GTM beacon for purchase tracking.